### DISCUSSION TITLE
Static Binary

### DISCUSSION THREAD

**Discussion by zebede22** [xmake-io/xmake#4667] [Q&A] [answered]
How to create a statically linked binary not library?

```
target("xxx")
    set_kind("binary")
    set_languages("c++20")
```

after calling `xmake` it generates binary with dynamically linked binary

**Comment by zebede22** [comment]
for clang we have -static for it, to create self contained binary

**Reply by waruqi** [reply, marked as answer]
```lua
add_ldflags("-static", {force = true})
```

**Reply by yh-sb** [reply]
Why do we need specify `{force = true}` here?
I tested it, there is no difference with `{force = true}` or without it.

**Reply by waruqi** [reply]
xmake automatically detects flags by default and ignores unsupported flags, but may fail to do so. Use force to avoid detection and force you to set it every time.

If you try to detect `-static` by compiling and linking, you may encounter a linking error and fail.
